assignments/arrays.js

```diff
@@ -63,32 +313,62 @@ let inventory = [{"id":1,"car_make":"Lincoln","car_model":"Navigator","car_year"
 
 // ==== Challenge 1 ====
 // The dealer can't recall the information for a car with an id of 33 on his lot. Help the dealer find out which car has an id of 33 by logging the car's year, make, and model in the console log provided to you below:
-console.log(`Car 33 is a *car year goes here* *car make goes here* *car model goes here*` );
+const arr = inventory
+for (let i = 0; i < arr.length; i++) {
+    if (arr[i].id === 33) {
+        console.log(`Car 33 is a ${arr[i].car_year} ${arr[i].car_make} ${arr[i].car_model}`);
+    }
+}
+//console.log(`Car 33 is a ${arr[i].car_year} ${arr[i].car_year} *car model goes here*`);
 
 // ==== Challenge 2 ====
 // The dealer needs the information on the last car in their inventory.  What is the make and model of the last car in the inventory?  Log the make and model into the console.
+
 let lastCar = 0;
-console.log();
+lastCar = arr[arr.length - 1]
+console.log(`last car is a ${lastCar.car_make} ${lastCar.car_model}`);
 
 // ==== Challenge 3 ====
 // The marketing team wants the car models listed alphabetically on the website. Sort all the car model names into alphabetical order and log the results in the console
-let carModels = [];
-console.log();
+
+let carModels = arr;
+carModels.sort(function (a, b) {
+    var nameA = a.car_model.toUpperCase();
+    var nameB = b.car_model.toUpperCase();
+    if (nameA < nameB) {
+        return -1;
+    }
+    if (nameA > nameB) {
+        return 1;
+    }
+})
+console.log(carModels);
 
 // ==== Challenge 4 ====
 // The accounting team needs all the years from every car on the lot. Create a new array from the dealer data containing only the car years and log the result in the console.
-let carYears = [];
-console.log();
+let carYears = []
+for (let i = 0; i < arr.length; i++) {
+    carYears.push(arr[i].car_year)
+}
+console.log(carYears);
 
 // ==== Challenge 5 ====
 // The car lot manager needs to find out how many cars are older than the year 2000. Using the carYears array you just created, find out how many cars were made before the year 2000 by populating the array oldCars and logging it's length.
 let oldCars = [];
-console.log(); 
+for (let i = 0; i < arr.length; i++) {
+    if (carYears[i] < 2000) {
+        oldCars.push(arr[i].carYears)
+    }
+}
+console.log(oldCars.length);
 
 // ==== Challenge 6 ====
 // A buyer is interested in seeing only BMW and Audi cars within the inventory.  Return an array that only contains BMW and Audi cars.  Once you have populated the BMWAndAudi array, use JSON.stringify() to show the results of the array in the console.
 let BMWAndAudi = [];
-console.log();
-
-
+for (let i = 0; i < arr.length; i++) {
+    if (carModels[i] || BMW || Audi) {
```

Comment:
Reconsider your conditional in your if statement here. `||` means "OR". So when you go,

`carModels[i] || 'BMW' || 'Audi'`

What are you saying?

Also, strings must have quotation marks. Without quotation marks, JS will believe you're trying to reference a variable, and will get mad when it can't find it.